### PR TITLE
fix: prevent record/pick from introducing non-existing properties

### PR DESCRIPTION
**Pull request by baoshan**
With [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) being set, `{ foo: undefined }` is no longer valid for type `{foo?: string}`. Although it is not part of the `strict` set, I wish we could respect the difference.

The current implementation of `record/pick` may introduce non-existing properties when such optional key(s) are picked:

```
pick("foo")({})
// { foo: undefined }
```

Proposed implementation is slightly modified from the code I committed to [`fp-ts-std`](https://github.com/samhh/fp-ts-std/commit/52156657df394da31b415d68a82fc660c76abf55).

I changed the signature to comply with [`Ramda`](https://ramdajs.com/docs/#pick) (and also `fp-ts-std`). A single array of keys is more composable generally IMHO.

Please let me know if this makes sense. Thank you.

### CODE REVIEW

File: record.ts [hunk 1]
```diff
@@ -104,18 +104,31 @@ export function omit<A, P extends keyof A>(
...
+ */
+export function pick<T, K extends keyof T>(
+  keys: readonly K[],
+): (record: T) => Pick<T, K> {
+  return (record) =>
+    keys.reduce(
+      (prev, key) => ({
+        ...prev,
+        ...(key in record ? { [key]: record[key] } : {}),
+      }),
```
Comment (baetheus) [inline record.ts:129]: This looks good but reduce operation here creates 2n-1 intermediate objects that are never used. While it's nice to remain pure, in cases like this I would prefer that we opt for space and performance to keep this operation light if it is used in a tight loop somewhere. Would you mind reverting to the for loop or using keys.forEach and mutably altering a single output object? ie.

```ts
export function pick<T, K extends keyof T>(
  keys: readonly K[],
): (record: T) => Pick<T, K> {
  return (record) =>
    keys.reduce(
      (prev, key) => {
        if (key in record) {
          prev[key] = record[key];
        }
        return prev;
      },
      {} as Pick<T, K>,
    );
}

// Or 

export function pick<T, K extends keyof T>(
  keys: readonly K[],
): (record: T) => Pick<T, K> {
  return (record) => {
    const output: Pick<T, K> = {};
    for (const key of keys) {
      if (key in record) {
        output[key] = record[key];
      }
    }
    return output;
  }
}
```

**Review by baetheus — CHANGES_REQUESTED** [review]
Thanks for looking into this!

I agree with the type changes that you've suggested but have a small implementation detail I'd like to see addressed. There is one change that I'm not sure about, and that is the change from requiring two or more keys in pick to requiring none. I don't remember the rational for this (it could just be something inherited from fp-ts), but I'd like a few days to think about it.

As always, thanks for the PR!

**Comment by baetheus** [comment]
Would you also mind running `deno fmt` to clear that check-format flag?

File: record.ts (hunk 1 above)
Comment (baoshan) [reply record.ts]: Sure. I vote for the vanilia `for` loop if you don’t mind.

**Comment by baoshan** [comment]
Please let me know if there’s something I left.

**Review by baetheus — APPROVED** [review]

**Pull request merged**